hi-ml-azure/src/health_azure/utils.py

```diff
@@ -1459,6 +1460,35 @@ def download_files_from_run_id(
     torch_barrier()
 
 
+def get_driver_log_file_text(run: Run, download_file: bool = True) -> Optional[str]:
+    """
+    Returns text stored in run log driver file.
+
+    :param run: Run object representing the current run.
+    :param download_file: Boolean flag that control if the log file should be downloaded from the run or not.
+    :return: Driver log file text if a file exists, `None` otherwise.
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
+
+        for log_file_path in VALID_LOG_FILE_PATHS:
+            if download_file:
+                run.download_files(
+                    prefix=str(log_file_path),
+                    output_directory=tmp_dir_name,
+                    append_prefix=False,
+                )
+            tmp_log_file_path = tmp_dir_name / log_file_path
+            if tmp_log_file_path.is_file():
+                return tmp_log_file_path.read_text()
+
+    files_as_str = ', '.join([f"'{str(log_file_path)}'" for log_file_path in VALID_LOG_FILE_PATHS])
```

Comment:
1. You don't need to use `str` in the f-string, the conversion will happen automatically.
2. You can lose the square brackets (some linter will raise errors about this)